### ISSUE TITLE
Store config re-encrypt encrypted values on save

### ISSUE THREAD

**Issue by bgkavinga** [magento/magento2#1223]
I have created a store config with backend type obscure
```xml
<field id="password" translate="label" type="obscure" sortOrder="6" showInDefault="1" showInWebsite="1" showInStore="1">
    <label>Password</label>
     <comment>
              <![CDATA[In Sandbox mode, use your Sandbox password.]]>
      </comment>
       <backend_model>Magento\Config\Model\Config\Backend\Encrypted</backend_model>
</field>
```
There are some other configs on the same section. Each time I save this section it encrypt the old value. Since the old value is already encrypted it is encrypting encrypted value. No way to decrypt and get original data. According to my investigations the problem lies in the following code snippet in Magento\Config\Model\Config\Backend\Encrypted
```php
/**
     * Encrypt value before saving
     *
     * @return void
     */
    public function beforeSave()
    {
        $value = (string)$this->getValue();
        // don't change value, if an obscured value came
        if (preg_match('/^\*+$/', $this->getValue())) {
            $value = $this->getOldValue();
        }
        if (!empty($value)) {
            $encrypted = $this->_encryptor->encrypt($value);
            if ($encrypted) {
                $this->setValue($encrypted);
            }
        }
    }
```

**Comment by bgkavinga** [comment]
Hello, Any update on this ? Is this true for you or it's happening just for me ? Am I doing any thing wrong here ?
Thanks
BG

**Comment by anupdugar** [comment]
@bgkavinga Thanks for reporting this issue. After investigation it seems that the value is getting encrypted again, even if you don't make any changes, like you pointed out.
We have created a ticket internally - MAGETWO-36930 to track and fix this issue.

**Comment by vpelipenko** [comment]
@bgkavinga, fix is available from 0.74.0-beta10. If you stile face with it, please, let us know.
